packages/ui/client/components/Navigation.vue

```diff
@@ -1,11 +1,19 @@
 <script setup lang="ts">
+import { injectCurrentModule, injectShowSummary, injectSummaryVisible } from '../composables/navigation'
+import { findById } from '../composables/client'
 import type { Task } from '#types'
 import { toggleDark } from '~/composables'
 import { files, runAll } from '~/composables/client'
 import { activeFileId } from '~/composables/params'
 
+const summaryVisible = injectSummaryVisible()
+const showSummary = injectShowSummary()
+const currentModule = injectCurrentModule()
```

Comment:
What's the reason for using inject/provide here? I would prefer to directly use global ref/computed for it. We are a SPA so we don't need to worry about the state mixing, I see inject/provide only add complexity but is not really helpful.